consul/resource_consul_service_test.go

```diff
@@ -16,122 +17,157 @@ func TestAccConsulService_basic(t *testing.T) {
-					testAccCheckConsulServiceValue("consul_service.app", "id", "google"),
-					testAccCheckConsulServiceValue("consul_service.app", "service_id", "google"),
-					testAccCheckConsulServiceValue("consul_service.app", "name", "google"),
-					testAccCheckConsulServiceValue("consul_service.app", "port", "80"),
-					testAccCheckConsulServiceValue("consul_service.app", "tags.#", "2"),
-					testAccCheckConsulServiceValue("consul_service.app", "tags.0", "tag0"),
-					testAccCheckConsulServiceValue("consul_service.app", "tags.1", "tag1"),
+					resource.TestCheckResourceAttr("consul_service.google", "id", "google"),
+					resource.TestCheckResourceAttr("consul_service.google", "address", "www.google.com"),
+					resource.TestCheckResourceAttr("consul_service.google", "node", "compute-google"),
+					resource.TestCheckResourceAttr("consul_service.google", "port", "80"),
+					resource.TestCheckResourceAttr("consul_service.google", "tags.#", "1"),
+					resource.TestCheckResourceAttr("consul_service.google", "tags.0", "tag0"),
 				),
 			},
 		},
 	})
 }
 
-func TestAccConsulService_extremove(t *testing.T) {
+func TestAccConsulService_basicModify(t *testing.T) {
 	resource.Test(t, resource.TestCase{
 		PreCheck:     func() {},
 		Providers:    testAccProviders,
 		CheckDestroy: testAccCheckConsulServiceDestroy,
 		Steps: []resource.TestStep{
 			resource.TestStep{
-				Config:             testAccConsulServiceConfig,
-				ExpectNonEmptyPlan: true,
+				Config: testAccConsulServiceConfigBasic,
 				Check: resource.ComposeTestCheckFunc(
-					testAccCheckConsulServiceExists(),
-					testAccCheckConsulServiceValue("consul_service.app", "address", "www.google.com"),
-					testAccCheckConsulServiceValue("consul_service.app", "id", "google"),
-					testAccCheckConsulServiceValue("consul_service.app", "service_id", "google"),
-					testAccCheckConsulServiceValue("consul_service.app", "name", "google"),
-					testAccCheckConsulServiceValue("consul_service.app", "port", "80"),
-					testAccCheckConsulServiceValue("consul_service.app", "tags.#", "2"),
-					testAccCheckConsulServiceValue("consul_service.app", "tags.0", "tag0"),
-					testAccCheckConsulServiceValue("consul_service.app", "tags.1", "tag1"),
-					testAccCheckConsulServiceDeregister("google"),
+					resource.TestCheckResourceAttr("consul_service.google", "id", "google"),
 				),
 			},
+			resource.TestStep{
+				Config: testAccConsulServiceConfigBasicNewTags,
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr("consul_service.google", "tags.#", "2"),
+					resource.TestCheckResourceAttr("consul_service.google", "tags.0", "tag0"),
+					resource.TestCheckResourceAttr("consul_service.google", "tags.1", "tag1"),
+				),
+			},
+			resource.TestStep{
+				Config: testAccConsulServiceConfigBasicAddress,
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr("consul_service.google", "address", "lb.google.com"),
+				),
+			},
+		},
```

Comment:
My only real nitpick with this test is that it checks that the tests modified what they were supposed to, but doesn't check that the things they weren't supposed to modify are still what we can expect. If there's a bug that causes a field to get unset in state, or set to a wrong value in state, on update this won't catch it.

Reply:
Sure, totally fair. Going to add some more checks.